homeassistant/components/elkm1/__init__.py

```diff
@@ -149,6 +149,11 @@ def _has_all_unique_prefixes(value):
 
     for index, conf in enumerate(hass_config[DOMAIN]):
         _LOGGER.debug("Importing elkm1 #%d - %s", index, conf[CONF_HOST])
+
+        # The update of the config entry is done in async_setup
+        # to ensure the entry if updated before async_setup_entry
+        # is called to avoid a situation where the user has to restart
+        # twice for the changes to take effect
```

Comment:
Another solution is to add a config entry update listener that reloads the entry.

Reply:
I tried this route, unfortunately it the elk system didn't handle it very well since we connect, disconnect, and then connect again in the middle of the sync that already started so it adds about 40s to the startup time since we disconnect in the middle of the first sync.
```
2020-03-29 21:16:02 INFO (MainThread) [elkm1_lib.elk] Connecting to ElkM1 at elks://192.168.209.56
2020-03-29 21:16:08 INFO (MainThread) [elkm1_lib.elk] Connected to ElkM1
2020-03-29 21:16:10 INFO (MainThread) [elkm1_lib.elk] Connecting to ElkM1 at elks://192.168.209.56
2020-03-29 21:16:18 INFO (MainThread) [elkm1_lib.elk] Connected to ElkM1
```

Reply:
Ok.